src/lib.rs

```diff
@@ -101,14 +102,19 @@ impl AuraedRuntime {
     pub async fn run(&self) -> Result<(), Box<dyn std::error::Error>> {
         // Manage the socket permission/groups first\
         let _ = fs::remove_file(&self.socket);
-        tokio::fs::create_dir_all(Path::new(&self.socket).parent().unwrap())
-            .await
-            .with_context(|| {
-                format!(
-                    "Failed to create directory for socket: {}",
-                    self.socket.display()
-                )
-            })?;
+        let sock_res = Path::new(&self.socket)
+            .parent()
+            .ok_or("unable to find socket path");
```

Comment:
You could have used the '?' operator here and avoid line 108-111 (can be iterated on later)